src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -43,10 +43,14 @@ public function __construct($data = null, $status = 200, $headers = array())
             $data = new \ArrayObject();
         }
 
+        $this->setDefaultEncodingOptions();
+        $this->setData($data);
+    }
+
+    protected function setDefaultEncodingOptions()
```

Comment:
Class has setter method, but it is no optimal
```php
public function setEncodingOptions($encodingOptions)
    {
        $this->encodingOptions = (int) $encodingOptions;

        return $this->setData(json_decode($this->data));
    }
```

When I need use not default encodingOptions code will work not optimally:
json_encode (constuctor -> setData) -> json_decode (setEncodingOptions) -> json_encode (setEncodingOptions -> setData)

I think better way could be fourth optional parameter to constructor with current hardcoded value encodingOptions as default, but expressions is disallow in function parameter.

Therefore, I see overwrite in the child class with the current implementation.

Do you have any ideas how to do this without inheritance?